alpha-commons/src/test/java/at/ac/tuwien/kr/alpha/commons/programs/ReificationHelperTest.java

```diff
@@ -0,0 +1,345 @@
+package at.ac.tuwien.kr.alpha.commons.programs;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import at.ac.tuwien.kr.alpha.api.programs.atoms.AggregateAtom;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.AggregateAtom.AggregateElement;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.AggregateAtom.AggregateFunctionSymbol;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.Atom;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.ComparisonAtom;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.ExternalAtom;
+import at.ac.tuwien.kr.alpha.api.programs.literals.Literal;
+import at.ac.tuwien.kr.alpha.api.terms.ArithmeticOperator;
+import at.ac.tuwien.kr.alpha.api.terms.ArithmeticTerm;
+import at.ac.tuwien.kr.alpha.api.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.api.terms.FunctionTerm;
+import at.ac.tuwien.kr.alpha.api.terms.Term;
+import at.ac.tuwien.kr.alpha.api.terms.VariableTerm;
+import at.ac.tuwien.kr.alpha.commons.Predicates;
+import at.ac.tuwien.kr.alpha.commons.atoms.Atoms;
+import at.ac.tuwien.kr.alpha.commons.comparisons.ComparisonOperators;
+import at.ac.tuwien.kr.alpha.commons.terms.Terms;
+import at.ac.tuwien.kr.alpha.commons.util.IntIdGenerator;
+
+public class ReificationHelperTest {
+
+	private Supplier<ConstantTerm<?>> newIdGenerator() {
+		return new Supplier<ConstantTerm<?>>() {
+
+			IntIdGenerator idGen = new IntIdGenerator(0);
+
+			@Override
+			public ConstantTerm<?> get() {
+				return Terms.newConstant(idGen.getNextId());
+			}
+
+		};
+	}
+
+	@Test
+	public void reifySymbolicConstant() {
+		ConstantTerm<String> constant = Terms.newSymbolicConstant("someConstant");
+		Supplier<ConstantTerm<?>> idGen = newIdGenerator();
+		ConstantTerm<?> reifiedId = idGen.get();
+		Set<BasicAtom> reified = new ReificationHelper(idGen).reifyConstantTerm(reifiedId, constant);
+		assertTrue(reified.contains(Atoms.newBasicAtom(Predicates.getPredicate("term_type", 2), reifiedId, Terms.newSymbolicConstant("constant"))));
```

Comment:
In case the `idGenerator` is moved into the `ReificationHelper` one can no longer know from the outside, which id will be generated for the reified term. This makes all these checks here more inconvenient, but could make all these tests not having to rely on a specific behaviour of the id generator. Could the `AnswerSetQuery` interface be of help to check for the presence of certain atoms? Of course, one would first need to add a functionality to generate an `AnswerSet` from a `Set<BasicAtom>`.

Reply:
Yes, that's a great idea. I renamed `AnswerSetQuery` to `AtomQuery` and adapted it slightly to be used here too.